pkg/cluster/internal/providers/docker/network.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	"sigs.k8s.io/kind/pkg/exec"
+)
+
+// TODO: we'll probably allow configuring this
+//
+// however currently picking a single network is equivilant to the previous
+// behavior *except* that we moved from the default bridge to a user defined
+// network because the default bridge is actually special versus any other
+// docker network and lacks the emebdded DNS
+//
+// for now this also makes it easier for apps to join the same network
+const fixedNetworkName = "kind"
+
+// ensureNetwork checks if docker network by name exists, if not it creates it
+func ensureNetwork(name string) error {
+	out, err := exec.Output(exec.Command(
+		"docker", "network", "ls",
+		"--filter=name=^"+name,
+		"--format={{.Name}}",
+	))
+	if err != nil {
+		return err
+	}
+	// network already exists
+	if string(out) == name+"\n" {
```

Comment:
using `docker inspect` we can avoid this

Reply:
I don't think so, but this isn't really a problem anyhow. I'm aware of inspect and chose this method.

This method is more precise, it only errors if docker cannot be called, and the value it returns is either "" or "name\n"

Reply:
that's a good point, we can differentiate between command errors and a not found network